FIAT/quadrature_schemes.py

```diff
@@ -329,6 +329,133 @@ def _kmv_lump_scheme(ref_el, degree):
         else:
             raise ValueError("Dimension not supported")
 
+    elif degree == 6:
+        if sd == 2:
+            x = list(ref_el.vertices)
+            alphas = [
+                5.00000000000000e-1,
+                8.29411811106452e-2,
+                2.68649695592714e-1,
+            ]
+            betas = [
+                4.68059729056814e-1,
+                7.93088545089875e-2,
+                3.92931636618867e-1,
+            ]
+            gammas = [
+                2.48172758709406e-1,
+                1.56582066033687e-1,
+            ]
+            deltas = [
+                6.99812197147049e-1,
+                2.43089592364562e-1,
+            ]
+            weights = [
+                5.35113520281665e-4,
+                4.29435346026293e-3,
+                3.02990950926060e-3,
+                3.16396316646563e-3,
+                2.43035184285235e-2,
+                1.66312091329395e-2,
+                3.42178857644876e-2,
+                1.73480160090330e-2,
+                1.98004044953264e-2,
+            ]
+
+            alpha = alphas[0]
+            x.extend(
+                [
+                    (alpha, alpha),
+                    (0.0, alpha),
+                    (alpha, 0.0),
+                ]
+            )  # edge midpoints (class 2 points)
+            alpha = alphas[1]
+            x.extend(
+                [
+                    (1 - alpha, alpha),
+                    (alpha, 1 - alpha),
+                    (0.0, 1 - alpha),
+                    (0.0, alpha),
+                    (alpha, 0.0),
+                    (1 - alpha, 0.0),
+                ]  # edge points (1st set of class 3 points)
```

Comment:
We should consider simplifying/generalizing this code. For instance I see that this code generates points on each individual edge for each degree. We could generate the points on the interval and then transform them with something like, 

```
pts = []
edge_points = generate_edge_kmv_points(degree)
top = ref_el.get_topololgy()
for edge_id in sorted(top[1])
   transform = ref_el.get_entity_transform(1, edge_id)
   pts.extend(transform(edge_points))
```
Also, do we know if the edge points coincide with any well known family of points, say GLL?

Reply:
Unfortunately, for degrees 3 and up, the edge points do not coincide with any well-known family of points, such as GLL, GRL, etc.. These elements are constructed by searching for configurations that satisfy specific geometric constraints, accuracy conditions, and positive weights, but no general rule is established yet. That’s why I’m introducing this new 6th-order element. It's entirely possible that, in the future, better elements for various orders could be discovered by refining or altering these search methods.